src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/QualityGuidelines/MarkMembersAsStatic.cs

```diff
@@ -111,15 +111,18 @@ void OnOperationBlockStart(OperationBlockStartAnalysisContext blockStartContext)
                         return;
                     }
 
-                    bool isInstanceReferenced = false;
+                    bool isInstanceReferenced = methodSymbol.AssociatedSymbol.IsIndexer();
 
-                    blockStartContext.RegisterOperationAction(operationContext =>
+                    if (!isInstanceReferenced)
```

Comment:
This might have performance benefits as we now not required to do a RegisterOperationAction for indexers.

Reply:
I think the code will be simpler if reverted to original form and just add the below if statement above this:

```csharp
if (methodSymbol.AssociatedSymbol.IsIndexer())
{
    return;
}
```

Reply:
Actually, can we just enhance the checks in `ShouldAnalyze` to check `!methodSymbol.AssociatedSymbol.IsIndexer()`?